### PR TITLE
[WIP] Add difficulty to track versions.

### PR DESCRIPTION
**Pull request by StefanMavrodiev**
Add difficulty rating to the track versions.

There is a missing functionality for versions edit. Thus difficulty can be set only upon creation. So this needs more work.

This should **not** be merged yet, just looking for review.

### CODE REVIEW

File: src/Entity/Track/Version.php [hunk 1]
```diff
@@ -168,4 +173,23 @@ public function getVotes(): int
...
+    {
+        return $this->difficulty;
+    }
+
+    /**
+     * Set track difficulty
+     *
+     * @param $difficulty
+     */
+    public function setDifficulty($difficulty): void
```
Comment (angelk) [inline src/Entity/Track/Version.php]: Please typehint the parameter `setDifficulty(string $difficulty = null)`

File: templates/include/mapVersion.html.twig [hunk 2]
```diff
@@ -28,6 +28,20 @@
             </button>
         {% endif %}
 
+        </td>
+        <td>
+            {% if version.difficulty == 'easiest' %}
+                <img src="{{ asset('images/difficulty/white-circle-48x48.png') }}">
```
Comment (angelk) [inline templates/include/mapVersion.html.twig]: What is the licence for these images?

File: script/migration/migrations/20190930200527_version_difficulty.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+use Phinx\Migration\AbstractMigration;
+
+class VersionDifficulty extends AbstractMigration
+{
+    public function up()
+    {
+        $this->query("
+            ALTER TABLE version
+            ADD difficulty 
+            ENUM('easiest', 'easy', 'more-difficult', 'very-difficult', 'extremely-difficult')
```
Comment (angelk) [inline script/migration/migrations/20190930200527_version_difficulty.php]: Why don't we use blue, green, black, 2black here?

File: assets/css/table.scss [hunk 4]
```diff
@@ -0,0 +1,9 @@
+.table {
```
Comment (angelk) [inline assets/css/table.scss]: Can we use separate class for this. `.table` is too generic

File: script/migration/migrations/20190930200527_version_difficulty.php (hunk 3 above)
Comment (StefanMavrodiev) [reply script/migration/migrations/20190930200527_version_difficulty.php]: I don't mind changing it, but I think it's better understood as it is. Many users doesn't know what 'black diamond' means, but when they see 'Very Difficult'... "yes, that track was very difficult".

If the level is set only by the admin (maybe will be better), then blue, green, etc is really better.

File: templates/include/mapVersion.html.twig (hunk 2 above)
Comment (StefanMavrodiev) [reply templates/include/mapVersion.html.twig]: These images are created by me.

File: script/migration/migrations/20190930200527_version_difficulty.php (hunk 3 above)
Comment (angelk) [reply script/migration/migrations/20190930200527_version_difficulty.php]: I think we should use Black diamin, green ... everywhere.
Users are already familiar with these difficulty system from other sites and bike parks.
Later we can include link to difficulty explanations.
For example https://bikepark.borovets-bg.com/?portfolio=machete
> DIFFICULTY: Blue

**Comment by StefanMavrodiev** [comment]
@angelk Please check 4b836956620394d74bd4a0cba6e28544d126b909

**Pull request merged**